### PR TITLE
[Part 3] Refactoring Mantis Master Client code into a Gateway interface that defines the spec and a curator based implementation that matches that.

### PR DESCRIPTION
**Pull request by sundargates**
Refactored all usages of the mantis mantis client to make use of the interface instead of the actual implementation.

### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable
- [ ] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServices.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+package io.mantisrx.server.master.client;
+
+import io.mantisrx.server.core.master.MasterDescription;
+import io.mantisrx.server.core.master.MasterMonitor;
+import io.mantisrx.shaded.com.google.common.util.concurrent.Service;
+import rx.Observable;
+
+public interface HighAvailabilityServices extends Service {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServices.java:23]: nit: add a high level comment on what this class is supposed to provide and be useful for

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import io.mantisrx.server.core.CoreConfiguration;
+import io.mantisrx.server.core.master.MasterDescription;
+import io.mantisrx.server.core.master.MasterMonitor;
+import io.mantisrx.server.core.zookeeper.CuratorService;
+import io.mantisrx.shaded.com.google.common.util.concurrent.AbstractIdleService;
+import lombok.extern.slf4j.Slf4j;
+import rx.Observable;
+
+@Slf4j
+public class HighAvailabilityServicesUtil {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java:27]: High level comment please.

File: mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java [hunk 3]
```diff
@@ -784,4 +796,31 @@ private Integer getSinkStageNumFromJsonResponse(String jobId, String response) t
 
         return Observable.merge(reconciliator.observables());
     }
+
+  private final AsyncHttpClient apiClient;
+
+  @Override
+  public CompletableFuture<Ack> updateStatus(Status status) {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java:803]: comment please. also context on why this was added.

**Review by calvin681 — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-client/src/test/java/io/mantisrx/server/master/client/MasterClientWrapperTest.java [hunk 4]
```diff
@@ -43,13 +45,16 @@
         zkProps.put("mantis.zookeeper.connectString", "100.67.80.172:2181,100.67.71.221:2181,100.67.89.26:2181,100.67.71.34:2181,100.67.80.18:2181");
         zkProps.put("mantis.zookeeper.leader.announcement.path", "/leader");
         zkProps.put("mantis.zookeeper.root", "/mantis/master");
+        zkProps.put("mantis.localmode", "false");
     }
 
     MasterClientWrapper clientWrapper = null;
 
     //@Before
```
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-client/src/test/java/io/mantisrx/server/master/client/MasterClientWrapperTest.java:53]: The before annotation is commented out?

**Pull request merged**